File: CoreGate/src/OpenSaur.CoreGate.Web/Features/Auth/Dtos/ConsentDtos.cs
namespace OpenSaur.CoreGate.Web.Features.Auth.Dtos;

public sealed record GetConsentDetailsRequest(
    string ReturnUrl);

public sealed record ConsentScopeItem(
    string Name,
    string Description);

public sealed record ConsentDetailsResponse(
    string ClientDisplayName,
    IReadOnlyList<ConsentScopeItem> Scopes,
    string ReturnUrl);

public sealed record ConsentDecisionRequest(
    string Decision,
    string ReturnUrl);

public sealed record ConsentDecisionResponse(
    string RedirectUrl);

File: CoreGate/src/OpenSaur.CoreGate.Web/Frontend/src/dtos/ConsentDtos.ts
export type ConsentScopeItem = {
  name: string;
  description: string;
};

export type ConsentDetailsResponse = {
  clientDisplayName: string;
  scopes: ConsentScopeItem[];
  returnUrl: string;
};

export type ConsentDecisionRequest = {
  decision: "accept" | "reject";
  returnUrl: string;
};

export type ConsentDecisionResponse = {
  redirectUrl: string;
};
